src/NuGet.Clients/NuGet.CommandLine/Commands/SearchCommand.cs

```diff
@@ -0,0 +1,208 @@
+            }
+            return packageSources;
+        }
+
+        public override async Task ExecuteCommandAsync()
+        {
+            ILogger logger = Console;
+            CancellationToken cancellationToken = CancellationToken.None;
+
+            SearchFilter searchFilter = new SearchFilter(includePrerelease: PreRelease);
+
+            IList<PackageSource> listEndpoints = GetEndpointsAsync();
+
+            foreach (PackageSource source in listEndpoints)
+            {
+                string target = source.Source;
+                string name = source.Name;
+
+                SourceRepository repository = Repository.Factory.GetCoreV3(target);
+                PackageSearchResource resource = await repository.GetResourceAsync<PackageSearchResource>();
+
+                if (resource is null)
+                {
+                    continue;
+                }
+
+                IEnumerable<IPackageSearchMetadata> results = await resource.SearchAsync(
+                    string.Join("+", Arguments).Trim().Replace(' ', '+'), // The arguments (query strings) are joined with '+' so that the queries are passed to the
+                                                                          // search query url with the correct format (eg. "logging extensions" --> "logging+extensions")
+                    searchFilter,
+                    skip: 0,
+                    take: Take,
+                    logger,
+                    cancellationToken);
+
+                Console.WriteLine(new string('=', 20));
+                Console.WriteLine($"Source: {source.Name}");
+
+                if (!results.Any())
+                {
+                    Console.WriteLine(new string('-', 20));
+                    Console.WriteLine("No results found.");
+                    Console.WriteLine(new string('-', 20) + "\n");
+                }
+                else
+                {
+                    if (Verbosity == Verbosity.Quiet)
+                    {
+                        System.Console.WriteLine($"Source: {source.Name}");
+                        System.Console.WriteLine(new string('-', 20));
+                    }
+                    PrintResults(results);
+                }
+            }
+        }
+
+        private void PrintResults(IEnumerable<IPackageSearchMetadata> results)
+        {
+            foreach (IPackageSearchMetadata result in results)
+            {
+                Console.WriteLine(new string('-', 20));
+
+                CultureInfo culture = CultureInfo.CurrentCulture;
+
+                string printBasicInfo = $"> {result.Identity.Id} | {result.Identity.Version.ToNormalizedString()}";
+
+                string downloads, printDownloads;
+
+                if (result.DownloadCount != null)
+                {
+                    downloads = string.Format(culture, "{0:N}", result.DownloadCount);
+                    printDownloads = $" | Downloads: {downloads.Substring(0, downloads.Length - 3)}";
```

Comment:
What's the 3? Is this safe for different CultureInfos?

Reply:
The Format string {0:N} returns a number with two decimal places, so I wanted to remove the ".00" from the end. As far as I'm aware, the only variation in this part of the string would be ",00" for some cultures, but I might be wrong about this.

Reply:
I think that's a fair assumption, as this indicates a default of 2 
https://docs.microsoft.com/en-us/dotnet/api/system.globalization.numberformatinfo.numberdecimaldigits?view=netcore-3.1